src/utils.ts

```diff
@@ -2082,21 +2184,139 @@ export class newTooltip {
   }
 }
 
-export class newAxis {
+const DASH_SELECTION_WINDOW = [7, 3];
+export class SelectionBox extends newRect {
+  public minVertex: Vertex;
+  public maxVertex: Vertex;
+
+  private _previousMin: Vertex;
+  private _previousMax: Vertex;
+
+  public leftUpdate: boolean = false;
+  public rightUpdate: boolean = false;
+  public upUpdate: boolean = false;
+  public downUpdate: boolean = false;
+  constructor(
+    public origin: Vertex = new Vertex(0, 0),
+    public size: Vertex = new Vertex(0, 0)
+  ) {
+    super(origin, size);
+    this.dashLine = DASH_SELECTION_WINDOW;
+    this.selectedStyle = this.clickedStyle = this.hoverStyle = this.fillStyle = "hsla(0, 0%, 100%, 0)";
+    this.lineWidth = 0.5
+  }
+
+  get isDefined(): boolean { return (this.minVertex != undefined && this.maxVertex != undefined) }
+
+  public setDrawingProperties(context: CanvasRenderingContext2D) {
+    super.setDrawingProperties(context);
+    context.lineWidth = (this.isHovered || this.isClicked) ? this.lineWidth * 2 : this.lineWidth;
+  }
+
+  public update(minVertex: Vertex, maxVertex: Vertex) {
+    this.minVertex = minVertex;
+    this.maxVertex = maxVertex;
+  }
+
+  public rubberBandUpdate(rubberBand: RubberBand, coordName: string) {
+    if (this.isDefined) {
+      this.minVertex[coordName] = rubberBand.minValue;
+      this.maxVertex[coordName] = rubberBand.maxValue;
+    }
+  }
+
+  public buildRectFromHTMatrix(plotOrigin: Vertex, plotSize: Vertex, HTMatrix: DOMMatrix) {
+    this.origin = this.minVertex.transform(HTMatrix);
+    this.size = this.maxVertex.transform(HTMatrix).subtract(this.origin);
+    this.insideCanvas(plotOrigin, plotSize);
+  }
+
+  private insideCanvas(drawOrigin: Vertex, drawSize: Vertex): boolean {
+    let isInside = true;
+    const downLeftCorner = this.origin;
+    const upRightCorner = downLeftCorner.add(this.size);
+    const upRightDiff = drawOrigin.add(drawSize).subtract(upRightCorner);
+    const downLeftDiff = downLeftCorner.subtract(drawOrigin);
+
+    if (upRightDiff.x < 0) this.size.x += upRightDiff.x
+    else if (upRightDiff.x > drawSize.x) this.size.x += upRightDiff.x - drawSize.x;
+
+    if (upRightDiff.y < 0) this.size.y += upRightDiff.y
+    else if (upRightDiff.y > drawSize.y) this.size.y += upRightDiff.y - drawSize.y;
+
+    if (downLeftDiff.x < 0) {
+      this.origin.x -= downLeftDiff.x;
+      this.size.x += downLeftDiff.x;
+    } else if (downLeftDiff.x > drawSize.x){
+      this.origin.x -= downLeftDiff.x - drawSize.x;
+      this.size.x += downLeftDiff.x - drawSize.x;
+    }
+    if (downLeftDiff.y < 0) {
+      this.origin.y -= downLeftDiff.y;
+      this.size.y += downLeftDiff.y;
+    } else if (downLeftDiff.y > drawSize.y){
+      this.origin.y -= downLeftDiff.y - drawSize.y;
+      this.size.y += downLeftDiff.y - drawSize.y;
+    }
+
+    return isInside
```

Comment:
isInside is never changed, so its always true ?

Reply:
Yes, removed